test/controllers/admins_controller.js

```diff
@@ -0,0 +1,151 @@
+        it('should return created status', done => {
+          chai
+            .request(server)
+            .post('/admin/users')
+            .set('Authorization', token)
+            .send({
+              user: {
+                password: '1234567890',
+                email: 'federico.dinucci@wolox.com.ar',
+                firstName: 'Federico',
+                lastName: 'Di Nucci'
+              }
+            })
+            .then(res => {
+              res.should.have.status(201);
+              User.findOne({ where: { email: 'federico.dinucci@wolox.com.ar' } }).then(
+                foundAdmin => foundAdmin.should.not.be.null
+              );
+              dictum.chai(res, 'Admin Creation');
+            })
+            .then(() => done());
+        });
+      });
+      context('when a required parameter is missing', () => {
+        it('should fail signing up', done => {
+          chai
+            .request(server)
+            .post('/admin/users')
+            .set('Authorization', token)
+            .send({
+              user: {
+                password: '1234567890',
+                email: 'federico.dinucci@wolox.com.ar',
+                firstName: 'Federico'
+              }
+            })
+            .catch(err => {
+              err.should.have.status(400);
+            })
+            .then(() => done());
+        });
+      });
+      context('when the password does not meet the requirements', () => {
+        it('should fail signing up', done => {
+          chai
+            .request(server)
+            .post('/admin/users')
+            .set('Authorization', token)
+            .send({
+              user: {
+                password: 'passwordwithoutnumbers',
+                email: 'federico.dinucci@wolox.com.ar',
+                firstName: 'Federico'
+              }
+            })
+            .catch(err => {
+              err.should.have.status(400);
+            })
+            .then(() => done());
+        });
+      });
+
+      context('when a user/admin with the same email exists', () => {
+        beforeEach(done => {
+          User.create({
+            password: '1234567890',
+            email: 'federico.dinucci@wolox.com.ar',
+            firstName: 'Federico',
+            lastName: 'Di Nucci'
+          }).then(() => done());
+        });
+        it('should upgrade him to admin', done => {
+          chai
+            .request(server)
+            .post('/admin/users')
+            .set('Authorization', token)
+            .send({
+              user: {
+                password: '1234567890',
+                email: 'federico.dinucci@wolox.com.ar',
+                firstName: 'Ignacio',
+                lastName: 'Torres'
+              }
+            })
+            .then(res => {
+              res.should.have.status(201);
+              let admin;
+              User.findOne({ where: { email: 'federico.dinucci@wolox.com.ar' } }).then(foundAdmin =>
```

Comment:
careful with this, since you are not returning the `User.findOne` promise it will probably go straight to done and never get enough time to validate `should.equal('ADMIN')`

Reply:
thats why as a rule of thumb I ALWAYS return all promises, regardless of where they are used